### PR TITLE
US20T04 Validation middleware for Recipe CRUD

### PR DESCRIPTION
**Pull request by yannduhamel**
Creation of a validation middleware for the POST request of the Recipe CRUD

### CODE REVIEW

**Review by Pierre-Voisin — APPROVED** [review]

**Review by enerisan — APPROVED** [review]

File: server/database/schema.sql [hunk 1]
```diff
@@ -48,16 +48,16 @@ CREATE TABLE menu (
...
     description TEXT NOT NULL,
     image TEXT NOT NULL,
-    date DATE NOT NULL,
-    is_favorite BOOLEAN,
-    vote INT,
-    set_up_time INT,
-    is_validated BOOLEAN,
-    user_id INT NOT NULL,
-    badge_id INT NOT NULL,
+    date DATE DEFAULT "1900-01-01",
```
Comment (ioayoub) [inline server/database/schema.sql:54]: prefer a "DEFAULT NOW()" here

**Review by ioayoub — CHANGES_REQUESTED** [review]

**Review by angelique888 — APPROVED** [review]

File: server/database/schema.sql [hunk 2]
```diff
@@ -48,16 +48,16 @@ CREATE TABLE menu (
...
+    is_favorite BOOLEAN DEFAULT false,
+    vote INT UNSIGNED DEFAULT 0,
+    set_up_time INT UNSIGNED NOT NULL,
+    is_validated BOOLEAN DEFAULT false,
+    user_id INT DEFAULT 1,
+    badge_id INT DEFAULT 1,
```
Comment (ioayoub) [inline server/database/schema.sql:60]: No, no and no ! 
Never put a default foreign key value !

**Review by ioayoub — CHANGES_REQUESTED** [review]

**Review by AnneDAMIEN — DISMISSED** [review]

File: server/database/schema.sql (hunk 2 above)
Comment (yannduhamel) [reply server/database/schema.sql]: I updated it

File: server/database/schema.sql [hunk 3]
```diff
@@ -48,16 +48,16 @@ CREATE TABLE menu (
...
-    set_up_time INT,
-    is_validated BOOLEAN,
-    user_id INT NOT NULL,
-    badge_id INT NOT NULL,
+    date DATE DEFAULT "1900-01-01",
+    is_favorite BOOLEAN DEFAULT false,
+    vote INT UNSIGNED DEFAULT 0,
+    set_up_time INT UNSIGNED NOT NULL,
+    is_validated BOOLEAN DEFAULT false,
+    user_id INT DEFAULT 1,
```
[the remark this answers is not in the document]
Comment (yannduhamel) [reply server/database/schema.sql]: I updated it

File: server/database/schema.sql (hunk 1 above)
Comment (yannduhamel) [reply server/database/schema.sql]: I changed it to a DATETIME DEFAULT NOW()

**Review by Pierre-Voisin — APPROVED** [review]

**Review by angelique888 — APPROVED** [review]

**Review by enerisan — APPROVED** [review]

**Review by AnneDAMIEN — APPROVED** [review]

**Pull request merged**